packages/runtime-core/src/components/Suspense.ts

```diff
@@ -321,7 +321,7 @@ function patchSuspense(
       triggerEvent(n2, 'onPending')
       // mount pending branch in off-dom container
       suspense.pendingBranch = newBranch
-      suspense.pendingId++
+      if (suspense.pendingId > 0) suspense.pendingId--
```

Comment:
the root cause is: the newBranch is cached in keepalive and its suspenseId not updated when pendingId++. so we should update the suspenseId of newBranch will be ok.

likely:
```javascript
if (
  parentComponent &&
  (parentComponent.vnode.type as any).__isKeepAlive &&
  newBranch.component
) {
  newBranch.component.suspenseId = suspense.pendingId
}
```